File: src/main/BomImportService/Services/PdfExtractor.cs
using System.Text;
using BomImportService.Interfaces;
using UglyToad.PdfPig;

namespace BomImportService.Services;

public class PdfExtractor : IPdfExtractor
{
    public string ExtractText(string pdfPath)
    {
        if (!File.Exists(pdfPath))
            throw new FileNotFoundException($"PDF not found: {pdfPath}");

        using var stream = File.OpenRead(pdfPath);
        return ExtractText(stream);
    }

    public string ExtractText(Stream pdfStream)
    {
        using var document = PdfDocument.Open(pdfStream);
        var sb = new StringBuilder();

        foreach (var page in document.GetPages())
        {
            var text = page.Text;

            if (!string.IsNullOrWhiteSpace(text))
            {
                sb.AppendLine($"--- Page {page.Number} ---");
                sb.AppendLine(text);
                sb.AppendLine();
            }
        }

        return sb.ToString();
    }
}

File: src/main/ariadne-py/ariadne/pdf_extractor.py
from __future__ import annotations

import pdfplumber


def extract_text_from_pdf(pdf_path: str) -> str:
    pages = []
    with pdfplumber.open(pdf_path) as pdf:
        for i, page in enumerate(pdf.pages):
            text = page.extract_text() or ""
            if text.strip():
                pages.append(f"--- Page {i + 1} ---\n{text}")
    return "\n\n".join(pages)


def extract_text_from_pdf_stream(stream) -> str:
    pages = []
    with pdfplumber.open(stream) as pdf:
        for i, page in enumerate(pdf.pages):
            text = page.extract_text() or ""
            if text.strip():
                pages.append(f"--- Page {i + 1} ---\n{text}")
    return "\n\n".join(pages)
